generator/php/generatorConfiguration.yaml

```diff
@@ -7,6 +7,14 @@ files:
     folder : test/Api
     templateType: SupportingFiles
     destinationFilename: ClientCredentialsClientTest.php
+  custom/OAuthClient.php.mustache:
+    folder : lib
+    templateType: SupportingFiles
+    destinationFilename: OAuthClient.php
+  custom/ClientCredentialsClient/Token.php.mustache:
```

Comment:
I'm not sure of the logic behind putting this in a dedicated ClientCredentialsClient folder. I think our tokens are the same regardless of authentication flow. Is there a reason why you added it to a Client Credentials directory?

Reply:
I understand your concern, unfortunately, yes, there is reason why I added it to a Client CredentialsClient directory: it is to preserve retrocompatibility.

It will become clear if we look at L263 of the file currently generated
https://github.com/criteo/criteo-api-marketingsolutions-php-sdk/blob/a4a59d1700020af7d1b8c4fed76fee0cd4048f56/lib/ClientCredentialsClient.php#L263
I don't know why we did that, but we change the namespace for the Token class.
Given that, in order to fix PSR-4 compatibility we have only two choices:
1. create the directory as I propose in this PR
2. or change the namespace, but that would break retrocompatibility (ie: clients who upgrade our packages would have their code broken, at least, until they fix their "import")

I'm afraid not solution is perfect, but I guess it's better if we suffer a bit rather if we make all our client suffer (at least we have clues on what is going on, whereas a client who would face such retrocompatibility issue would likely be clueless about why it breaks (unless they look in our code or in our changelog))

Reply:
Got it, makes sense.